Put retention days on Cloud Watch logs. with the TRaceback

Here is the Policy that I'm running periodically. I'm getting throttling Errors.

policies:
  - name: Custodian-loggroup-retention
    resource: log-group
    description: |
        Checks log groups weekely and sets the log retention for log groups that doesn't have log retention set.
    mode:
       type: periodic
       schedule: "cron(0 12 ? * 2 *)"
       role: CustodianLambdaExecutionRole
       packages:
          - boto3
          - botocore
    filters:
      - type: value
         key: retentionInDays
         value: absent
    actions:
       - type: retention
         days: 400

**Here is the traceback.**

[DEBUG] 2020-02-10T16:00:25.670Z fc012410-e0d7-49a5-a52f-c8a996ffb8e8 metric:ResourceCount Count:820 policy:custodian-loggroup-retention restype:log-group scope:policy
[DEBUG] 2020-02-10T16:00:25.670Z fc012410-e0d7-49a5-a52f-c8a996ffb8e8 metric:PolicyException Count:1 policy:custodian-loggroup-retention restype:log-group
[DEBUG] 2020-02-10T16:00:25.670Z fc012410-e0d7-49a5-a52f-c8a996ffb8e8 metric:ApiCalls Count:110 policy:custodian-loggroup-retention restype:log-group
[ERROR] 2020-02-10T16:00:25.670Z fc012410-e0d7-49a5-a52f-c8a996ffb8e8 Error while executing policy
Traceback (most recent call last):
File "/var/task/c7n/policy.py", line 320, in run
results = a.process(resources)
File "/var/task/c7n/resources/cw.py", line 201, in process
retentionInDays=days)
File "/var/task/botocore/client.py", line 276, in _api_call
return self._make_api_call(operation_name, kwargs)
File "/var/task/botocore/client.py", line 586, in _make_api_call
raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ThrottlingException) when calling the PutRetentionPolicy operation (reached max retries: 4): Rate exceeded

--- a/c7n/resources/cw.py
+++ b/c7n/resources/cw.py
@@ -196,7 +196,8 @@ class Retention(BaseAction):
         client = local_session(self.manager.session_factory).client('logs')
         days = self.data['days']
         for r in resources:
-            client.put_retention_policy(
+            self.manager.retry(
+                client.put_retention_policy,
                 logGroupName=r['logGroupName'],
                 retentionInDays=days)
 

